demo/midgard-node/src/database/utils/error.ts

```diff
@@ -0,0 +1,145 @@
+import { Database } from "@/services/database.js";
+import { SqlClient, SqlError } from "@effect/sql";
+import { Data, Effect } from "effect";
+
+export class DatabaseError extends Data.TaggedError("DatabaseError")<{
+  readonly message: string;
+  readonly operation:
+    | "select"
+    | "insert"
+    | "update"
+    | "delete"
+    | "createTable"
+    | "other";
```

Comment:
Similar to my other comment for `TransactionError`, I think keeping this sum type at type level will provide a better DevEx, i.e. instead of "hiding" this sum type under `operation`, we can have `DBSelectError`, `DBCreateError`, etc.